clients/agentic-sandbox-client-python/agentic_sandbox/sandbox.py

```diff
@@ -0,0 +1,188 @@
+        self.server_port: int = 8888  # Assuming a default port
+
+        try:
+            config.load_incluster_config()
+        except config.ConfigException:
+            config.load_kube_config()
+
+        self.custom_objects_api = client.CustomObjectsApi()
+
+    def is_ready(self) -> bool:
+        """Returns True if the sandbox is created and ready for communication."""
+        return self.port_forward_process is not None
+
+    def __enter__(self) -> 'Sandbox':
+        """Creates the SandboxClaim resource and waits for the Sandbox to become ready."""
+        self.claim_name = f"sandbox-claim-{os.urandom(4).hex()}"
+        manifest = {
+            "apiVersion": f"{CLAIM_API_GROUP}/{CLAIM_API_VERSION}",
+            "kind": "SandboxClaim",
+            "metadata": {"name": self.claim_name},
+            "spec": {"sandboxTemplateRef": {"name": self.template_name}}
+        }
+
+        print(f"Creating SandboxClaim: {self.claim_name}...")
+        self.custom_objects_api.create_namespaced_custom_object(
+            group=CLAIM_API_GROUP,
+            version=CLAIM_API_VERSION,
+            namespace=self.namespace,
+            plural=CLAIM_PLURAL_NAME,
+            body=manifest
+        )
+
+        w = watch.Watch()
+        print("Watching for Sandbox to become ready...")
+        for event in w.stream(
+            func=self.custom_objects_api.list_namespaced_custom_object,
+            namespace=self.namespace,
+            group=SANDBOX_API_GROUP,
+            version=SANDBOX_API_VERSION,
+            plural=SANDBOX_PLURAL_NAME,
+            field_selector=f"metadata.name={self.claim_name}",
+            timeout_seconds=180
+        ):
+            sandbox_object = event['object']
+            status = sandbox_object.get('status', {})
+            conditions = status.get('conditions', [])
+            is_ready = False
+            for cond in conditions:
+                if cond.get('type') == 'Ready' and cond.get('status') == 'True':
+                    is_ready = True
+                    break
+
+            if is_ready:
+                self.sandbox_name = sandbox_object['metadata']['name']
+                w.stop()
+                print(f"Sandbox {self.sandbox_name} is ready.")
+                break
+
+        if not self.sandbox_name:
+            self.__exit__(None, None, None)
+            raise TimeoutError("Sandbox did not become ready within the 180-second timeout period.")
+
+        print(f"Starting port-forwarding for sandbox {self.sandbox_name}...")
+        self.port_forward_process = subprocess.Popen(
+            [
+                "kubectl", "port-forward",
+                f"pod/{self.sandbox_name}",
+                f"{self.server_port}:{self.server_port}",
+                "-n", self.namespace
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        time.sleep(3)  # Give port-forward a moment to establish
+
```

Comment:
Using `time.sleep(3)` to wait for the port-forward to be established is not reliable. A more robust approach would be to poll the port to see if it's open and listening before proceeding. You could try to connect to the port in a loop with a timeout.